### ISSUE TITLE
Omega (Ω) symbol appears instead of @ on GB keyboard

### ISSUE THREAD

**Issue by nbolton** [symless/synergy#62]
**Imported issue:**

* Author: nick.bolton.uk (Google user)
* Date: 2009-05-25 00:00:00
* Legacy ID: 71

**What steps will reproduce the problem?**  
1. Client: Ubuntu 9.04  
2. Server: Windows Vista SP1  
3. Press right shift and the [@ '] key  
  
**What is the expected output? What do you see instead?**  
  
Expect: @  
See: Ω

**Comment by matthewbloch** [comment]
FWIW, this happened on the last nightly build for me (Debian/jessie client, Windows server), and the xmodmap fix worked.

**Comment by the-wes** [comment]
See also https://github.com/synergy/synergy/issues/4352

**Comment by ivebeenlinuxed** [comment]
I won't be "buying" synergy while this bug is around.

**Comment by mwelchuk** [comment]
Seeing this bug on Debian Jessie (as both client and server).
Both machines have LANG=en_GB.UTF-8.
Both are amd64 (x86_64).
Both are running version 1.4.16-1.

**Comment by Lanvar** [comment]
I am experiencing the same problem, server is a windows 7 (64bit) system and the client is a Linux Mint (17.2) box. Extremely frustrating.

Any word of a fix on this bug yet? It has been open over a year....

**Comment by nosnickid** [comment]
I'd also love a fix.  Seems like it's been open over 6 years... 

"
Imported issue:

Author: nick.bolton.uk (Google user)
Date: 2009-05-25 00:00:00
"

**Comment by ivebeenlinuxed** [comment]
I find it very annoying that this bug is not fixed, yet there is talk about adding "serial keys" in in the next milestone. All I've seen in these milestones is "flashy features", without any graft fixing existing bugs.

Maybe I should start packaging Synergy myself for my business needs... I've offered money and that didn't work...

**Comment by wrapper** [comment]
Here to add a +1 to needing a fix for this

**Comment by bobdvb** [comment]
+1 to having just run into this.

Server: Windows 7
Client: Ubuntu 14.04 configured to enGB.UTF-8

**Comment by kjbweb** [comment]
+1 same here

Server Windows 7
Client: Linux Mint 17.2

A 6 year old bug! Wow... :-/

**Comment by StreetGuru** [comment]
Ticking "Use System keyboard layout" in Advanced settings of the "Ibus Preferences" as solved the problem for me.

**Comment by prabjohns-atex** [comment]
Same for me, +1
(Paid up Donater to Synergy)

**Comment by rrthomas** [comment]
Sounds good; however, I'm using uim, not iBus, and see no similar option in its prefs.

**Comment by dannystaple** [comment]
I'm still reproducing this.
Server: Windows 8 with British Laptop keyboard
Client: Fedora 23 with UK Keyboard configured.
Versions: Synergy Pro 1.7.6 at both ends.
The Xmodmap workaround appears to work for now.

**Comment by niczero** [comment]
The stated workaround used to work for me, and then it stopped working.  If you are one of the many casualties who are seeing omega rather than at, and you're using UTF, the workaround is as follows.
  1. Check that the client's locale is as compatible as possible with the server.
  ```
  locale
  ```

  2. Get the current settings of your xmodmap.
  ```
  xmodmap -pke | grep -w 24
  ```

  3. Write a client startup script that sets the keyboard and modifies the sequence from #2.  (You probably only need to replace the one 'omega' value with 'at'.)
  ```
  synergyc --daemon $SERVER_IP &>>/var/tmp/synergy.log
  setxkbmap gb
  xmodmap -e 'keycode 24 = q Q q Q at at at'
  ```

The key message is that the number of values you set must match the number previously set, or you will see silent failure.

**Comment by nbolton** [comment]
@XinyuHou @nlyan We should solve this bug soon.

**Comment by XinyuHou** [comment]
Could someone test this nightly please?
http://symless.com/nightly?filter=issue2765-spacebar-issue-beta-fbd4363

**Comment by XinyuHou** [comment]
> This was previously reported in issue #62 a couple years ago (#62)
> 
> The workaround within the issue works great for me to get around this but, I have pasted it below for your convenience. I guess you can try changing the symbol in the script if you have issues with other symbols for other keys such as "\" or "/" or "|".
> 
> For reference I'm running 08th August's Nightly build. Server is Win 10 and client is Ubuntu 16.04.
> 
> #!/bin/bash
> 
> #Start ssh
> 
> killall ssh
> 
> sleep 1
> 
> ssh -f -N -L localhost:24800::24800 @
> 
> #Start Synergyclient
> 
> killall synergyc
> 
> sleep 1
> 
> synergyc localhost
> 
> #Workaround synergy ?/@ key issue
> 
> sleep 1
> 
> xmodmap -e "keycode 53 = x X greater greater greater greater"
> 
> xmodmap -e "keycode 52 = z Z less less less less"
> 
> xmodmap -e "keycode 24 = q Q at at at at"

**Comment by brunovianarezende** [comment]
I just tested with ubuntu 14.04 server and client. This problem don't happen anymore.

**Comment by nlyan** [comment]
@brunovianarezende What Synergy version did you test with?

**Comment by brunovianarezende** [comment]
version 1.8.3 both in client and server.

**Comment by nlyan** [comment]
I'm closing this as fixed in 1.8.3. If anyone still has this, or a similar, problem after this version please open a new issue.
